main.cpp

```diff
@@ -40,43 +40,66 @@ std::vector<std::string> splittedString(std::string text, char delimiter){
   return splittedStrings;
 }
 
-
-static Character* parseUnit(const std::string& fileName){
+static Character* createCharacter(const std::string& fileContent){
   std::string name = "";
   int health = 0;
   int damage = 0;
 
-  std::fstream characterDataFile;
-  characterDataFile.open(fileName);
- 
+  for(const auto& line : splittedString(fileContent, '\n'))
+  {
+    std::vector<std::string> splittedLine = splittedString(line, ':');
+    if(splittedLine.front().find("name") != std::string::npos){
+      name = splittedLine.back();
+      name = name.substr(2, name.length() - 4);
+    }
+    else  if(splittedLine.front().find("hp") != std::string::npos){
+      std::string inputHealth =  splittedLine.back();
+      inputHealth = inputHealth.substr(0, inputHealth.length()-1);
+      health = std::stoi(inputHealth);
+    }
+    else  if(splittedLine.front().find("dmg") != std::string::npos){
+      std::string damageHealth =  splittedLine.back();
+      damage = std::stoi(damageHealth);
```

Comment:
Nagyon érzékeny a kód a json fájl formázására. Minden pozíció fixen "be van égetve" a kódba, miközben egy json fájlban meglehetősen szabadon lehet fehér karaktereket beszúrni/törölni, arról nem beszélve, hogy a fájl formailag hibás is lehet. Legalább néhány ilyen hibára fel kellene készíteni a parzert.